### ISSUE TITLE
bastille upgrade cannot upgrade from release to itself

### ISSUE THREAD

**Issue by lifepillar** [BastilleBSD/bastille#300]
I have recently updated my host from 12.1-RELEASE to 12.2-RELEASE. Today, I have tried for the first time to upgrade my jails, currently based on 12.1-RELEASE. For that, I have successfully run `bastille update 12.1-RELEASE`, then I have rebooted. Now, when I try:

    bastille upgrade 12.1-RELEASE 12.2-RELEASE  

or, equivalently:

    freebsd-update -b /data/bastille/releases/12.1-RELEASE -r 12.2-RELEASE upgrade

I get this error:

    freebsd-update: Cannot upgrade from 12.2-RELEASE to itself

Any idea why? `freebsd-version` in any of my jails still outputs `12.1-RELEASE-p12`. My Bastille configuration is essentially the default one, except that I have disabled ZFS and changed `bastille_prefix` to `/data/bastille`. The original release was bootstrapped with `bastille bootstrap 12.1-RELEASE`.

**Bastille and FreeBSD version (paste ``bastille -v && freebsd-version -kru`` output)**

    0.7.20200714
    12.2-RELEASE-p1
    12.2-RELEASE-p1
    12.2-RELEASE-p2

**How did you install bastille? (port/pkg/git)**

port

**Comment by xaque208** [comment]
I see this too.  Confused how to upgrade.

**Comment by johnjgsmith** [comment]
Getting the same issue here.

**Comment by JRGTH** [comment]
Hello, I fix has been applied in this PR: ed8ff8bd802c3373b81a30a15a1057009ab71f11 to properly upgrade base RELEASE, the update is a two step as expected but I will briefly explain it here.

**Usage:**
`bastille upgrade 12.1-RELEASE 12.2-RELEASE`
`bastille upgrade 12.1-RELEASE 12.2-RELEASE -f`
`bastille upgrade 12.1-RELEASE install`
`bastille upgrade 12.1-RELEASE install -f`
Where `-f|--force` is useful for: _Force freebsd-update fetch to proceed in the case of an unfinished upgrade._

**Current issue example:**
```
root@nas-mserver: ~# bastille upgrade 12.1-RELEASE 12.2-RELEASE -f
src component not installed, skipped
freebsd-update: Cannot upgrade from 12.2-RELEASE to itself
root@nas-mserver: ~# bastille cmd test freebsd-version
[test]:
12.1-RELEASE-p10
```

**After patch applied examples**

**Upgrade Step #1 (I optionally used --force option here):**
```
root@nas-mserver: ~# bastille upgrade 12.1-RELEASE 12.2-RELEASE -f
src component not installed, skipped
Looking up update.FreeBSD.org mirrors... 3 mirrors found.
Fetching metadata signature for 12.1-RELEASE from update1.freebsd.org... done.
Fetching metadata index... done.
Fetching 1 metadata patches. done.
Applying metadata patches... done.
Fetching 1 metadata files... done.
Inspecting system... done.

The following components of FreeBSD seem to be installed:
kernel/generic world/base world/doc

The following components of FreeBSD do not seem to be installed:
kernel/generic-dbg world/base-dbg world/lib32 world/lib32-dbg

Does this look reasonable (y/n)? y

Fetching metadata signature for 12.2-RELEASE from update1.freebsd.org... done.
Fetching metadata index... done.
Fetching 1 metadata patches. done.
Applying metadata patches... done.
Fetching 1 metadata files... done.
Inspecting system... done.
Preparing to download files... done.
The following files will be removed as part of updating to
12.2-RELEASE-p2:
/etc/rc.d/abi
/usr/include/c++/v1/experimental/any
/usr/include/c++/v1/experimental/chrono
/usr/include/c++/v1/experimental/numeric
/usr/include/c++/v1/experimental/optional
...................
........................
..............................
/var/db/etcupdate/current/etc/termcap.small
/var/db/mergemaster.mtree
/var/db/services.db
/var/yp/Makefile.dist
To install the downloaded upgrades, run "/usr/sbin/freebsd-update install".

Please run 'bastille upgrade 12.1-RELEASE install' to finish installing updates.
```

***Upgrade Step #2:*
```
root@nas-mserver: ~# bastille upgrade 12.1-RELEASE install
src component not installed, skipped
Installing updates...
Kernel updates have been installed.  Please reboot and run
"/usr/sbin/freebsd-update install" again to finish installing updates.
root@nas-mserver: ~# bastille upgrade 12.1-RELEASE install
src component not installed, skipped
Installing updates...Scanning /mnt/storage/bastille/releases/12.1-RELEASE/usr/share/certs/blacklisted for certificates...
Scanning /mnt/storage/bastille/releases/12.1-RELEASE/usr/share/certs/trusted for certificates...
 done.
root@nas-mserver: ~# bastille cmd test freebsd-version
[test]:
12.2-RELEASE-p2
```

**Notes:**
The base RELEASE will remains named as 12.1-RELEASE despite it has been upgraded to 12.2-RELEASE as expected, and this should remains this way since there bay be the case an user already has been bootstrapped the 12.2-RELEASE.

Alternatively an user can also bootstrap an 12.2-RELEASE, then stop the jail then manually edit config files to make the thinjail to point to the new RELEASE path, IIRC I've created a command called `change` in order to make this configuration dynamically but never got much of attention but will look for again.

As expected UPGRADING a base RELEASE will affect all child containers depending on said RELEASE, so be warned.

Regards

**Comment by bmac2** [comment]
@JRGTH put this in a PR in JAN 2021.  Closing as fixed.
